dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -431,11 +457,17 @@ private List<Host> findPaginatedSitesFromDB(final User user, final int limit, fi
     @Override
     public List<Host> findAllFromCache(final User user,
             final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
-        Set<Host> cachedSites = hostCache.getAllSites();
+        Set<Host> cachedSites = hostCache.getAllSites(respectFrontendRoles);
         if(null == cachedSites){
-            final List<Host> allFromDB = findAllFromDB(user, respectFrontendRoles);
-            hostCache.addAll(allFromDB);
-            cachedSites = hostCache.getAllSites();
+            final List<Host> allFromDB = findAllFromDB(APILocator.systemUser(),
+                    true, false,false);
+            final List<Host> allFromDBLive = findAllFromDB(APILocator.systemUser(),
+                    true, true,false);
```

Comment:
to many flag this would be smoother to have  findAllFromDB(APILocator.systemUser(),
                    new Options().withIncludeSystemHost().withRetrieveLiveVersion().build());
                    
                 Any with call turn on the flag in true, otherwise assumes false

Reply:
Added Options to set flags for findAllFromDB method